SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -205,6 +363,164 @@ def evaluate_integral(self, receiver_location, components):
             ]
         )
 
+    def _forward(self, densities):
+        """
+        Forward model the fields of active cells in the mesh on receivers.
+
+        Parameters
+        ----------
+        densities : (n_active_cells) array
+            Array containing the densities of the active cells in the mesh, in
+            g/cc.
+
+        Returns
+        -------
+        (nD,) array
+            Always return a ``np.float64`` array.
+        """
+        # Gather active nodes and the indices of the nodes for each active cell
+        active_nodes, active_cell_nodes = self._get_active_nodes()
+        # Allocate fields array
+        fields = np.zeros(self.survey.nD, dtype=self.sensitivity_dtype)
+        # Compute fields
+        index_offset = 0
+        for components, receivers in self._get_components_and_receivers():
+            n_components = len(components)
+            n_elements = n_components * receivers.shape[0]
+            for i, component in enumerate(components):
+                kernel_func = CHOCLO_KERNELS[component]
+                conversion_factor = _get_conversion_factor(component)
+                vector_slice = slice(
```

Comment:
I believe that you are properly taking care of the ordering for multiple components but it could be worth adding a check to make sure that it is consistent (here and the sensitivity as well).

Reply:
Nice catch. This was one of trickiest parts of this PR: making sure that we are filling the right rows of the sensitivity matrix and the right elements of the fields array.

The tests pass several components of the acceleration and the tensor, so we are actually testing this bits. In case these slices were wrongly defined, those tests should be failing.

Is there any other test/check we could be running?

Reply:
I see where those tests are now, just didn't pop out to me on the first pass